planning/mission_planner/src/lanelet2_plugins/default_planner.cpp

```diff
@@ -186,21 +194,114 @@ PlannerPlugin::MarkerArray DefaultPlanner::visualize(const HADMapRoute & route)
   return route_marker_array;
 }
 
-bool DefaultPlanner::is_goal_valid(const geometry_msgs::msg::Pose & goal) const
+visualization_msgs::msg::MarkerArray DefaultPlanner::visualize_debug_footprint(
+  tier4_autoware_utils::LinearRing2d goal_footprint) const
+{
+  visualization_msgs::msg::MarkerArray msg;
+  auto marker = tier4_autoware_utils::createDefaultMarker(
+    "map", rclcpp::Clock().now(), "goal_footprint", 0, visualization_msgs::msg::Marker::LINE_STRIP,
+    tier4_autoware_utils::createMarkerScale(0.05, 0.0, 0.0),
+    tier4_autoware_utils::createMarkerColor(0.99, 0.99, 0.2, 1.0));
+  marker.lifetime = rclcpp::Duration::from_seconds(2.5);
+
+  marker.points.push_back(
+    tier4_autoware_utils::createPoint(goal_footprint[0][0], goal_footprint[0][1], 0.0));
+  marker.points.push_back(
+    tier4_autoware_utils::createPoint(goal_footprint[1][0], goal_footprint[1][1], 0.0));
+  marker.points.push_back(
+    tier4_autoware_utils::createPoint(goal_footprint[2][0], goal_footprint[2][1], 0.0));
+  marker.points.push_back(
+    tier4_autoware_utils::createPoint(goal_footprint[3][0], goal_footprint[3][1], 0.0));
+  marker.points.push_back(marker.points.front());
+
+  msg.markers.push_back(marker);
+
+  return msg;
+}
+
+bool DefaultPlanner::check_goal_footprint(
+  const lanelet::ConstLanelet & current_lanelet,
+  const lanelet::ConstLanelet & combined_prev_lanelet,
+  const tier4_autoware_utils::Polygon2d & goal_footprint, double & next_lane_length,
+  const double search_margin)
+{
+  std::vector<tier4_autoware_utils::Point2d> points_intersection;
+
+  // check if goal footprint is in current lane
+  boost::geometry::intersection(
+    goal_footprint, combined_prev_lanelet.polygon2d().basicPolygon(), points_intersection);
+  if (points_intersection.empty()) {
+    return true;
+  }
+  points_intersection.clear();
```

Comment:
I think there is no need to clear here.

Reply:
I think it is necessary for some situations, for example if the vehicle is not inside the goal lanelet, points_intercestion will go to the search with non-empty, which means that even if there is no intersection in the new search, we will assume that there is an intersection because of these remaining points.

Reply:
Before evaluating points_intersection on line 251, we overwrite points_intersection on line 249, 250.
Therefore I thought we do not need to clear points_intersection. Do I understand something wrong?
![image](https://user-images.githubusercontent.com/20228327/204178525-83c3dcef-fc96-45f4-bcba-240c898e6c6e.png)

Reply:
@takayuki5168 hi, ```boost::geometry::intersection``` function doesn't overwrite points, it pushes back intersection points to existing ```point_intersection``` vector. Thus, if ```points_intersection``` go to the for loop section with non-empty, its falsely affects algorithm.

Reply:
I got it. Thanks for your explanation.